code/modules/research/bepis.dm

```diff
@@ -0,0 +1,231 @@
+	withdraw_value = banking_amount
+	if(withdraw_value > banked_cash)
+		say("Cannot withdraw more than stored funds. Aborting.")
+	else
+		banked_cash -= withdraw_value
+		new /obj/item/holochip(src.loc, withdraw_value)
+		say("Withdrawing [withdraw_value] credits from the chamber.")
+	update_icon_state()
+	return
+
+/obj/machinery/rnd/bepis/proc/calcsuccess()
+	var/turf/dropturf = get_turf(pick(view(1,src)))
+	var/gauss_major = 0
+	var/gauss_minor = 0
+	var/gauss_real = 0
+	if(!dropturf) //Check to verify the turf exists and the reward isn't lost somehow.
+		dropturf = drop_location()
+	gauss_major = (gaussian(major_threshold, std) - negative_cash_offset)	//This is the randomized profit value that this experiment has to surpass to unlock a tech.
+	gauss_minor = (gaussian(minor_threshold, std) - negative_cash_offset)	//And this is the threshold to instead get a minor prize.
+	gauss_real = (gaussian(banked_cash, std*inaccuracy_percentage) + positive_cash_offset)	//this is the randomized profit value that your experiment expects to give.
+	say("Real: [gauss_real]. Minor: [gauss_minor]. Major: [gauss_major].")
+	flick("chamber_flash",src)
+	update_icon_state()
+	banked_cash = 0
+	if((gauss_real >= gauss_major) && (SSresearch.techweb_nodes_experimental.len > 0)) //Major Success.
+		say("Experiment concluded with major success. New technology node discovered on technology disc.")
+		new /obj/item/disk/tech_disk/major(dropturf,1)
+		if(SSresearch.techweb_nodes_experimental.len == 0)
+			say("Expended all available experimental technology nodes. Resorting to minor rewards.")
+		return
+	if(gauss_real >= gauss_minor) //Minor Success.
+		var/reward_number = 1
+		say("Experiment concluded with partial success. Dispensing compiled research efforts.")
+		reward_number = rand(1,2)
+		if(reward_number == 1)
+			new /obj/item/stack/circuit_stack/full(dropturf)
+		if(reward_number == 2)
+			new /obj/item/airlock_painter/decal(dropturf)
+		return
+	if(gauss_real <= -1)	//Critical Failure
+		say("ERROR: CRITICAL MACHIME MALFUNCTI- ON. CURRENCY IS NOT CRASH. CANNOT COMPUTE COMMAND: 'make bucks'") //not a typo, for once.
+		new /mob/living/simple_animal/deer(dropturf, 1)
+		use_power(MACHINE_OVERLOAD * power_saver) //To prevent gambling at low cost and also prevent spamming for infinite deer.
+		return
+	//Minor Failure
+	error_cause = pick("attempted to sell grey products to American dominated market.","attempted to sell gray products to British dominated market.","placed wild assumption that PDAs would go out of style.","simulated product #76 damaged brand reputation mortally.","simulated business model resembled 'pyramid scheme' by 98.7%.","product accidently granted override access to all station doors.")
+	say("Experiment concluded with zero product viability. Cause of error: [error_cause]")
+	return
+
+/obj/machinery/rnd/bepis/update_icon_state()
+	if(panel_open == TRUE)
+		icon_state = "chamber_open"
+		return
+	if((powered == TRUE) && is_operational())
```

Comment:
If you were keeping this, just if(powered) would do the same thing. The opposite would be if(!powered) for false.